### ISSUE TITLE
Is this actually a supervised training for disease classification?

### ISSUE THREAD

**Issue by cherise215** [Jason-Qiu/Transfer_Knowledge_from_Language_to_Electrocardiography#1]
First of all, great work! But I am a bit confused when reading your code. It seems like, during training, you need to compute two losses: the OT loss and a supervised loss for the disease classification task. code: https://github.com/Jason-Qiu/Transfer_Knowledge_from_Language_to_Electrocardiography/blob/525266e5f35a016c309fe07e3fe9da17c66144ae/utils.py#L26

If this is the case, why do you report the result as zero-shot cardiovascular disease detection in the paper?  Can you explain it in more detail?

**Comment by Jason-Qiu** [comment]
Thanks for your interest in our work! Because we finetune the LLM for a non-classification objective, the finetuning only aims at outputting better textual descriptions of the input ECG signals. We call it zero-shot because the model is not trained under a supervised classification task, where the embeddings are not finetuned for disease detection. After finetuning, we can directly use the transformed embeddings in the classification task, but the model actually did not learn any disease detection-specific knowledge in the finetuning step, so we call it zero-shot here. Please let us know if there is any other questions :)

**Comment by JIELINQIU** [comment]
The classification labels are not used in the training phase. For more details, please see our updated version here: https://aclanthology.org/2023.findings-eacl.33.pdf (Section 3 Problem Formulation, Model Architecture, Downstream Applications etc). Thanks.

**Comment by Jason-Qiu** [comment]
The classification labels are not used in the training phase. For more details, please see our updated version here: https://aclanthology.org/2023.findings-eacl.33.pdf (Section 3 Problem Formulation, Model Architecture, Downstream Applications etc). Thanks.

**Comment by cherise215** [comment]
In your paper, you said "in addition to the
traditional cross-entropy loss. "  Isn't that loss include the classification labels?

**Comment by wan363** [comment]
Have you run your code? I think your code is shit! Your code have just done n-points window filtering to ECG signal during preprocess，some running results in two "ipynb " files's  is wrong，you use labels in training phase but you still call it zero-shot，your code just use 13 time domain features and 9 frequency features. "utils.py " import "load" from "evaluate.py" but "evaluate.py" don't have "load". Do you think it makes sense to import "utils. py" from "translate. py", import "load" from "evaluate. py", and read "translations_864.csv" from "evaluate. py".

**Comment by wan363** [comment]
Have you run your code? I think your code is shit! Your code have just done n-points window filtering to ECG signal during preprocess，some running results in two "ipynb " files's is wrong，you use labels in training phase but you still call it zero-shot，your code just use 13 time domain features and 9 frequency features. "utils.py " import "load" from "evaluate.py" but "evaluate.py" don't have "load". Do you think it makes sense to import "utils.py" in "translate.py", import "load" from "evaluate.py" in "utils. py", and read "translations_864.csv" in "evaluate. py".
